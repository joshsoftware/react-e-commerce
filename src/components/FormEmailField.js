import React from 'react';
import FormField from './FormField';
import { setEmail } from '../actions/formActions';
import PropTypes from 'prop-types';

let email = {
  field: 'exampleEmail',
  labelText: 'Email',
  type: 'email',
  name: 'email',
  placeholder: 'example@company.com'
};
const FormEmailField = (props) => {
  email = {
    ...email,
    value: props.email,
    onChange: (evt) => {
      console.log(evt);
      props.dispatch(setEmail(evt));
    },
    invalid: props.emailError !== null ? true : false,
    message: props.emailError
  };
  return <FormField formfield={email} />;
};

export default FormEmailField;

FormEmailField.propTypes = {
  email: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  emailError: PropTypes.string
};
