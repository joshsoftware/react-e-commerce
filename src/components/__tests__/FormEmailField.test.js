import FormEmailField from '../FormEmailField';
import { render } from '@testing-library/react';
import React from 'react';

describe('FormEmailField Component', () => {
  let email = '';
  const dispatch = () => {};
  it('Must return FormEmailField Component', () => {
    const { asFragment } = render(<FormEmailField email={email} dispatch={dispatch} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
