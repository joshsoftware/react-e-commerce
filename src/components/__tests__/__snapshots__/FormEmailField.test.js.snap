// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormEmailField Component Must return FormEmailField Component 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <label
      class=""
      for="exampleEmail"
    >
      Email
    </label>
    <input
      aria-invalid="true"
      class="is-invalid form-control"
      name="email"
      placeholder="example@company.com"
      type="email"
      value=""
    />
    <div
      class="invalid-feedback"
    />
  </div>
</DocumentFragment>
`;
